### ISSUE TITLE
Weights Not Loading for text_branch in Training Script

### ISSUE THREAD

**Issue by WangHaoyuuu** [LAION-AI/CLAP#158]
I am experiencing an issue with a training script for an audio-visual model where the text_branch components are not loading any pre-trained weights as expected. The unloaded components include all layers of text_branch, such as word embeddings, position embeddings, attention layers, and all associated weights and biases. Is this necessary?

here is the warning：
![7eaae4339b849a33e8d9256d252b26a](https://github.com/user-attachments/assets/1f1fb07a-4549-4180-b749-08ea7df8738a)

text_branch.embeddings.word_embeddings.weight    Unloaded
text_branch.embeddings.position_embeddings.weight        Unloaded
text_branch.embeddings.token_type_embeddings.weight      Unloaded
text_branch.embeddings.LayerNorm.weight          Unloaded
text_branch.embeddings.LayerNorm.bias    Unloaded
text_branch.encoder.layer.0.attention.self.query.weight          Unloaded
text_branch.encoder.layer.0.attention.self.query.bias    Unloaded
text_branch.encoder.layer.0.attention.self.key.weight    Unloaded
text_branch.encoder.layer.0.attention.self.key.bias      Unloaded
text_branch.encoder.layer.0.attention.self.value.weight          Unloaded
text_branch.encoder.layer.0.attention.self.value.bias    Unloaded
text_branch.encoder.layer.0.attention.output.dense.weight        Unloaded
text_branch.encoder.layer.0.attention.output.dense.bias          Unloaded
text_branch.encoder.layer.0.attention.output.LayerNorm.weight    Unloaded
text_branch.encoder.layer.0.attention.output.LayerNorm.bias      Unloaded
text_branch.encoder.layer.0.intermediate.dense.weight    Unloaded
text_branch.encoder.layer.0.intermediate.dense.bias      Unloaded
text_branch.encoder.layer.0.output.dense.weight          Unloaded
text_branch.encoder.layer.0.output.dense.bias    Unloaded

Here is my shell script:
#!/bin/bash

CUDA_VISIBLE_DEVICES=1 python -m training.main \
    --save-frequency 5 \
    --save-top-performance 3 \
    --save-most-recent \
    --dataset-type="webdataset" \
    --datasetpath='/home/ubuntu/AudioLDM-training-finetuning/data/dataset' \
    --precision="fp32" \
    --batch-size=32 \
    --lr=1e-4 \
    --wd=0.0 \
    --epochs=45 \
    --workers=1 \
    --use-bn-sync \
    --amodel HTSAT-tiny \
    --tmodel roberta \
    --warmup 3200 \
    --datasetnames "Clotho" \
    --datasetinfos "train" \
    --top-k-checkpoint-select-dataset="Clotho-test" \
    --top-k-checkpoint-select-metric="mAP@10" \
    --logs 'logs' \
    --seed 3407 \
    --gather-with-grad \
    --optimizer "adam" \
    --data-filling "repeatpad" \
    --data-truncating "rand_trunc" \
    --pretrained-audio '/home/ubuntu/AudioLDM-training-finetuning/data/checkpoints/HTSAT-fullset-imagenet-tiny-map=0.467.ckpt'

Here are my evaluation results:
![b6aa058625936a6a48c43a9ff5c145a](https://github.com/user-attachments/assets/c0b0f216-3b8b-4b81-9022-baeb5ed2b648)

It is very bad.

Could you please help me?

**Comment by zjsong** [comment]
> I am experiencing an issue with a training script for an audio-visual model where the text_branch components are not loading any pre-trained weights as expected. The unloaded components include all layers of text_branch, such as word embeddings, position embeddings, attention layers, and all associated weights and biases. Is this necessary?
> 
> here is the warning： ![7eaae4339b849a33e8d9256d252b26a](https://private-user-images.githubusercontent.com/95087066/348116190-1f1fb07a-4549-4180-b749-08ea7df8738a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMzQ4NTEsIm5iZiI6MTcyMTEzNDU1MSwicGF0aCI6Ii85NTA4NzA2Ni8zNDgxMTYxOTAtMWYxZmIwN2EtNDU0OS00MTgwLWI3NDktMDhlYTdkZjg3MzhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDEyNTU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwMzc4OGUwZWY3MWVhMGZhYWUzZTY5MWNjMDhkODVjMzg1YzEyZjZmM2QxYTdmMjlhNWM3NDg5M2M0ZjRhZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AH6OYHIbMDDCMZ2cW12L39mt1pog9_uJOG2YVHszTlA)
> 
> text_branch.embeddings.word_embeddings.weight Unloaded text_branch.embeddings.position_embeddings.weight Unloaded text_branch.embeddings.token_type_embeddings.weight Unloaded text_branch.embeddings.LayerNorm.weight Unloaded text_branch.embeddings.LayerNorm.bias Unloaded text_branch.encoder.layer.0.attention.self.query.weight Unloaded text_branch.encoder.layer.0.attention.self.query.bias Unloaded text_branch.encoder.layer.0.attention.self.key.weight Unloaded text_branch.encoder.layer.0.attention.self.key.bias Unloaded text_branch.encoder.layer.0.attention.self.value.weight Unloaded text_branch.encoder.layer.0.attention.self.value.bias Unloaded text_branch.encoder.layer.0.attention.output.dense.weight Unloaded text_branch.encoder.layer.0.attention.output.dense.bias Unloaded text_branch.encoder.layer.0.attention.output.LayerNorm.weight Unloaded text_branch.encoder.layer.0.attention.output.LayerNorm.bias Unloaded text_branch.encoder.layer.0.intermediate.dense.weight Unloaded text_branch.encoder.layer.0.intermediate.dense.bias Unloaded text_branch.encoder.layer.0.output.dense.weight Unloaded text_branch.encoder.layer.0.output.dense.bias Unloaded
> 
> Here is my shell script: #!/bin/bash
> 
> CUDA_VISIBLE_DEVICES=1 python -m training.main --save-frequency 5 --save-top-performance 3 --save-most-recent --dataset-type="webdataset" --datasetpath='/home/ubuntu/AudioLDM-training-finetuning/data/dataset' --precision="fp32" --batch-size=32 --lr=1e-4 --wd=0.0 --epochs=45 --workers=1 --use-bn-sync --amodel HTSAT-tiny --tmodel roberta --warmup 3200 --datasetnames "Clotho" --datasetinfos "train" --top-k-checkpoint-select-dataset="Clotho-test" --top-k-checkpoint-select-metric="mAP@10" --logs 'logs' --seed 3407 --gather-with-grad --optimizer "adam" --data-filling "repeatpad" --data-truncating "rand_trunc" --pretrained-audio '/home/ubuntu/AudioLDM-training-finetuning/data/checkpoints/HTSAT-fullset-imagenet-tiny-map=0.467.ckpt'
> 
> Here are my evaluation results: ![b6aa058625936a6a48c43a9ff5c145a](https://private-user-images.githubusercontent.com/95087066/348115936-c0b0f216-3b8b-4b81-9022-baeb5ed2b648.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMzQ4NTEsIm5iZiI6MTcyMTEzNDU1MSwicGF0aCI6Ii85NTA4NzA2Ni8zNDgxMTU5MzYtYzBiMGYyMTYtM2I4Yi00YjgxLTkwMjItYmFlYjVlZDJiNjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDEyNTU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5Nzc2NWUwODliYTgzNjUyYmI0NDk1N2Q5NTc3ZjdhZjQ3NzQ2OGUzY2IzOWVmNzM1ZDkzMWJkYzE3NmVjN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pXEvmRCdrHbVgzP2LcBtxq_s85oRSl7_CCoCXslqGao)
> 
> It is very bad.
> 
> Could you please help me?

It seems like that the aforementioned weights would be loaded successfully when using the updated version ([1.1.6](https://pypi.org/project/laion-clap/)). 
![1](https://github.com/user-attachments/assets/e7da9957-5780-4d9e-a097-c037b3f57cd8)

However, the warning is still there:
![2](https://github.com/user-attachments/assets/32670399-4fb4-427d-bf06-4a97b85ffa4e)

Not sure if this matters.

**Comment by oguzalpb** [comment]
I bumped into the same issue.
Have you solved the problem @WangHaoyuuu ?
